src/index.spec.ts

```diff
@@ -4,87 +4,143 @@
 
 import { HttpClient } from '@eppo/js-client-sdk-common';
 import { POLL_INTERVAL_MS, POLL_JITTER_PCT } from '@eppo/js-client-sdk-common/dist/constants';
-import { IExperimentConfiguration } from '@eppo/js-client-sdk-common/dist/dto/experiment-configuration-dto';
-import { EppoValue } from '@eppo/js-client-sdk-common/dist/eppo_value';
-import * as md5 from 'md5';
+import { Flag, VariationType } from '@eppo/js-client-sdk-common/dist/interfaces';
 import * as td from 'testdouble';
 import mock from 'xhr-mock';
 
 import {
   IAssignmentTestCase,
-  ValueTestType,
+  SubjectTestCase,
   readAssignmentTestData,
-  readMockRacResponse,
+  readMockUFCResponse,
+  MOCK_UFC_RESPONSE_FILE,
+  OBFUSCATED_MOCK_UFC_RESPONSE_FILE,
 } from '../test/testHelpers';
 
 import { EppoLocalStorage } from './local-storage';
 import { LocalStorageAssignmentCache } from './local-storage-assignment-cache';
 
-import { IAssignmentLogger, IEppoClient, getInstance, init } from './index';
+import { EppoJSClient, IAssignmentLogger, IEppoClient, getInstance, init } from './index';
+
+const flagEndpoint = /flag_config\/v1\/config*/;
+
+function getTestAssignments(
+  testCase: IAssignmentTestCase,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  assignmentFn: any,
+  obfuscated = false,
+): { subject: SubjectTestCase; assignment: string | boolean | number | null | object }[] {
+  const assignments: {
+    subject: SubjectTestCase;
+    assignment: string | boolean | number | null | object;
+  }[] = [];
+  for (const subject of testCase.subjects) {
+    const assignment = assignmentFn(
+      subject.subjectKey,
+      testCase.flag,
+      testCase.defaultValue,
+      subject.subjectAttributes,
+      obfuscated,
+    );
+    assignments.push({ subject: subject, assignment: assignment });
+  }
+  return assignments;
+}
+
+function validateTestAssignments(
+  assignments: {
+    subject: SubjectTestCase;
+    assignment: string | boolean | number | object | null;
```

Comment:
`null` should not be allowed here.

Reply:
@vpai Not sure if you missed this one or if the common repo also needs to be updated. Everything else looks good!